types/meteor/test/meteor-tests.ts

```diff
@@ -673,9 +672,48 @@ namespace MeteorTests {
     var userId = instance.data.userId;
 
     var data = Template.currentData();
-    var data = Template.parentData(1);
+    var data2 = Template.parentData();
+    var data3 = Template.parentData(2);
     var body = Template.body;
 
+    const Template2 = Template as TemplateStaticTyped<
+        { foo: string },
+        'newTemplate2',
+        {
+            state: ReactiveDict<{ bar: number }>;
+            getFooBar(): string;
+        }
+    >;
+
+    const newTemplate2 = Template2.newTemplate2;
```

Comment:
Can we assert that it's actually a `Blaze.Template<...>` and not `any`? Similarly, it'd be great to assert most of the operations below, just to be sure.

Reply:
Yes, we can assert it this way. Because `TemplateStaticTyped` is a helper type and has to be used explicitly by the developer. It's just a better type cast. :) I don't know the Blaze code internally. But for most use cases this should work.

Using it saves a lot of unnecessary casts and declarations. And makes the code more readable, maintainable and type-safe.

It has no effects on existing code that us `Template["..."]` or `Template.`.

Reply:
I guess we misunderstood :stuck_out_tongue: I meant adding a few `// $ExpectType X` here and there, just to make sure that none of the `any`s will "leak" to the public interface - it'd be a shame to lose these types in the future.